### ISSUE TITLE
helm test --logs cannot find the test pod

### ISSUE THREAD

**Issue by franck102** [helm/helm#9264]
Output of `helm version`:
`version.BuildInfo{Version:"v3.4.1", GitCommit:"c4e74854886b2efe3321e185578e6db9be0a6e29", GitTreeState:"dirty", GoVersion:"go1.15.4"}`

Output of `kubectl version`:
```
Client Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.8", GitCommit:"35dc4cdc26cfcb6614059c4c6e836e5f0dc61dee", GitTreeState:"clean", BuildDate:"2020-06-26T03:43:27Z", GoVersion:"go1.13.9", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"18+", GitVersion:"v1.18.9-eks-d1db3c", GitCommit:"d1db3c46e55f95d6a7d3e5578689371318f95ff9", GitTreeState:"clean", BuildDate:"2020-10-20T22:18:07Z", GoVersion:"go1.13.15", Compiler:"gc", Platform:"linux/amd64"}
```

Cloud Provider/Platform (AKS, GKE, Minikube etc.): 
AWS & Docker for Desktop

Running this command:
`helm test --logs dms-timescaledb`

yields

```
LAST DEPLOYED: Thu Jan 21 17:35:28 2021
NAMESPACE: dmp-monitoring
STATUS: deployed
REVISION: 6
TEST SUITE:     dms-timescaledb-tests
Last Started:   Fri Jan 22 07:33:30 2021
Last Completed: Fri Jan 22 07:33:33 2021
Phase:          Failed

Error: unable to get pod logs for dms-timescaledb-tests: pods "dms-timescaledb-tests" not found
```
The actual pod that the job created is
`dms-timescaledb-tests-bhqkr                              0/1     Error              0          5m26s`

The test job spec is a Job:

```
apiVersion: batch/v1
kind: Job
metadata:
  name: "{{ .Chart.Name }}-tests"
  labels:
    app.kubernetes.io/name: {{ .Chart.Name }}
    app.kubernetes.io/instance: {{ .Release.Name }}
    app.kubernetes.io/component: timescale-tests
    app.kubernetes.io/version: {{ .Chart.AppVersion | quote }}
    app.kubernetes.io/managed-by: {{ .Release.Service }}
  annotations:
    "helm.sh/hook": test
    "helm.sh/hook-delete-policy": before-hook-creation
spec:
  backoffLimit: 0
  template:
    metadata:
      name: "{{ .Chart.Name }}-tests"
    spec:
      ...
      containers:
        - name: "{{ .Chart.Name }}-tests"
          image: ...
          ...
      restartPolicy: Never
```

**Comment by jdolitsky** [comment]
Bug confirmed. Appears related to using a `Job` (vs. `Pod`), but perhaps also related to https://github.com/helm/helm/issues/8018

**Comment by franck102** [comment]
> Bug confirmed. Appears related to using a `Job` (vs. `Pod`), but perhaps also related to #8018

I believe I tried a Pod as well, with the same result (not 100% sure though).

**Comment by prqgdbc** [comment]
Can confirm I still see this when using a pod.

**Comment by deepflame** [comment]
We got the error when using the Log-Option with the Elastic-Chart.
When we removed 

```
 "helm.sh/hook-delete-policy": hook-succeeded
```

it was working fine.

Whole test here:
https://github.com/elastic/helm-charts/blob/master/elasticsearch/templates/test/test-elasticsearch-health.yaml#L8

**Comment by reitzig** [comment]
Related: #9098, fix proposed in #10603

**Comment by reitzig** [comment]
Unstale
